src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -376,29 +381,48 @@ object SvPileup extends LazyLogging {
    * @param seg2 the second alignment segment
    * @param maxWithinReadDistance the maximum distance between segments if they are from the same read
    * @param maxBetweenReadDistance the maximum distance between segments if they are from different reads
+   * @param dict the sequence dictionary to use for determining if a contig is circular
    */
   def isIntraContigBreakpoint(seg1: AlignedSegment,
                               seg2: AlignedSegment,
                               maxWithinReadDistance: Int,
-                              maxBetweenReadDistance: Int): Boolean = {
+                              maxBetweenReadDistance: Int,
+                              dict: SequenceDictionary): Boolean = {
     require(seg1.range.refIndex == seg2.range.refIndex)
-
+    
     // The way aligned segments are generated for a template, if we have all the reads in the expected orientation
-    // the segments should all come out on the same strand.  Therefore any difference in strand is odd.  In addition
-    // any segment that "moves backwards" down the genome is odd, as genome position and read position should increase
-    // together.
-    if (seg1.positiveStrand != seg2.positiveStrand) true
-    else if (seg1.positiveStrand && seg2.range.start < seg1.range.end) true
-    else if (!seg1.positiveStrand && seg1.range.start < seg2.range.start) true
+    // the segments should all come out on the same strand. Therefore any difference in strand is odd.
+    val positive_strand = seg1.positiveStrand
+    if (positive_strand != seg2.positiveStrand) {
+      return true
+    }
```

Comment:
We try to avoid use of explicit `returns` in scala.  Could you reformat as something like:

```scala
positive_strand != seg2.positiveStrand || {
  // rest of function
}
```